### ISSUE TITLE
Inconsistent readURL use.

### ISSUE THREAD

**Issue by robert-bo-davis** [blockstack/gaia#116]
While experimenting with gaia-hub running against various providers, I noticed that the readURL configuration variable is used inconsistently. 

In the disk driver, the readRUL is used without modification besides verifying a trailing slash:
```
this.readURL = config.readURL
if (this.readURL.slice(-1) !== '/') {
  // must end in /
  this.readURL = `${this.readURL}/`
}
```

In az it is prefixed with `https://` and has 
```
getReadURLPrefix () {
  if (this.readURL) {
    return `https://${this.readURL}/${this.bucket}/`
  }
  return `https://${this.accountName}.blob.core.windows.net/${this.bucket}/`
}
```

In gc and S3, it has `https://` prepended, but nothing else

gc:
```
getReadURLPrefix () {
  if (this.readURL) {
    return `https://${this.readURL}/`
  }
  return `https://storage.googleapis.com/${this.bucket}/`
}
```

S3
```
getReadURLPrefix(): string {
  if (this.readURL) {
    return `https://${this.readURL}/`
  }
  return `https://${this.bucket}.s3.amazonaws.com/`
}
```

The inconsistency makes configuration confusing. I would suggest that `readURL` always expect a full url including bucket name.

I'm happy to fix this if you'll provide guidance on the desired behavior.

Thanks,
Bo

**Comment by robert-bo-davis** [comment]
I see the discussion of a formal spec in #80. I'm happy to work on implementing that spec if it has been finalized as well.

Thanks,
Bo

**Comment by kantai** [comment]
Hi -- thanks for raising this issue @robert-bo-davis --

This is indeed a problem, and I think for the most consistency, it might be better to lift this code entirely out of the driver model. That is, the configuration setting for the `readUrl` is applied by, say, `server.js`, instead of implemented by each driver themselves.

Will open a PR for this now.

**Comment by kantai** [comment]
This was fixed in 2.1.0
